### PR TITLE
Testing table of contents

### PR DESCRIPTION
**Pull request by musangowope**
- Add end to end tests for TOC and TOCController components
resolves https://github.com/laws-africa/la-web-components/issues/33

### CODE REVIEW

File: src/components/table-of-contents/table-of-contents.e2e.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      `<la-table-of-contents items='${JSON.stringify(sampleItems)}' title-filter='Item One'></la-table-of-contents>`
+    );
+
+    const renderedItems = await page.findAll('la-table-of-contents la-toc-item');
+    expect(renderedItems[0].className.includes('excluded')).toBe(false);
+    expect(renderedItems[1].className.includes('excluded')).toBe(true);
+    expect(renderedItems[2].className.includes('excluded')).toBe(true);
+  });
+
+  it('should filter items on change of titleFilter prop', async () => {
```
Comment (longhotsummer) [inline src/components/table-of-contents/table-of-contents.e2e.tsx:58]: Good edge cases to include when doing tests on string inputs are:

* empty string
* string that is just spaces
* leading and trailing spaces
* case sensitivity

Comment (longhotsummer) [reply src/components/table-of-contents/table-of-contents.e2e.tsx:58]: In terms of behaviour, I propose: search is case insensitive (I think it is already) and leading and trailing spaces are stripped from inputs before they're used.

File: src/components/table-of-contents/data.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[
```
Comment (longhotsummer) [inline src/components/table-of-contents/data.json:1]: Test data is usually called a fixture, so that it's clear it's testing related. So `fixtures.json`?

File: src/components/table-of-contents/table-of-contents.e2e.tsx (hunk 1 above)
Comment (musangowope) [reply src/components/table-of-contents/table-of-contents.e2e.tsx:46]: added test cases for the above

File: src/components/table-of-contents/data.json (hunk 2 above)
Comment (musangowope) [reply src/components/table-of-contents/data.json]: done

**Review by longhotsummer — APPROVED** [review]
Great! Nice that we found and fixed two bugs with these tests.
